app/components/webview/DAppWebView.tsx

```
@@ -192,6 +197,73 @@ export const DAppWebView = memo(forwardRef((props: DAppWebViewProps, ref: Forwar
                 return;
             }
 
+            // Wallet API
+            if (props.useWalletAPI && parsed.data.name.startsWith('wallet')) {
+                if (Platform.OS !== 'ios') {
+                    warn('Wallet API is only available on iOS');
+                    return;
+                }
+                const method = parsed.data.name.split('.')[1] as 'isEnabled' | 'checkIfCardIsAlreadyAdded' | 'canAddCard' | 'addCardToWallet';
+
+                switch (method) {
+                    case 'isEnabled':
+                        (async () => {
```

Promise failure is not handled here